session.go

```diff
@@ -1611,12 +1611,25 @@ func (idxs indexSlice) Swap(i, j int)      { idxs[i], idxs[j] = idxs[j], idxs[i]
 
 func simpleIndexKey(realKey bson.D) (key []string) {
 	for i := range realKey {
+		var vi int
 		field := realKey[i].Name
-		vi, ok := realKey[i].Value.(int)
-		if !ok {
+
```

Comment:
For compatibility on 32 bit architectures I assume.

`int` is an `int64` on 64 bit systems.